server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -382,6 +423,7 @@ public void onResponse(ShardSnapshotResult shardSnapshotResult) {
                         snapshotStatus.generation()
                     );
                 }
+                snapshotShutdownProgressTracker.decNumberOfShardSnapshotsInProgress(snapshotStatus.getStage());
```

Comment:
I'd rather we did this with an `ActionListener#runBefore` just to make it clearer to readers that it always happens.